precovery/main.py

```diff
@@ -61,18 +64,22 @@ def precover_many(
 
 
 def precover_worker(
-    orbit: AdamOrbits,
+    orbit: Orbits,
     database_directory: str,
     tolerance: float = 1 / 3600,
     start_mjd: Optional[float] = None,
     end_mjd: Optional[float] = None,
     window_size: int = 7,
     allow_version_mismatch: bool = False,
     datasets: Optional[set[str]] = None,
-) -> Tuple[PrecoveryCandidatesQv, FrameCandidatesQv]:
+    propagator: Optional[Type[Propagator]] = None,
+) -> Tuple[PrecoveryCandidates, FrameCandidates]:
     """
     Wraps the precover function to return the orbit_id for mapping.
     """
+
+    # initialize our propagator
+    propagator_instance = propagator() if propagator is not None else None
```

Comment:
We should be passing a class here too and only initializing inside the db.precover function I think. Perhaps we should change the argument name to `propagator_class` ?

Reply:
I adopted `propagator_class` throughout